cmake/scripts/common/ModuleHelpers.cmake

```diff
@@ -66,4 +72,75 @@ macro(SETUP_BUILD_VARS)
   if(VERBOSE)
     message(STATUS "${MODULE}_URL: ${${MODULE}_URL}")
   endif()
-endmacro()
\ No newline at end of file
+
+  # unset all build_dep_target variables to insure clean state
+  unset(CMAKE_ARGS)
+  unset(PATCH_COMMAND)
+  unset(CONFIGURE_COMMAND)
+  unset(BUILD_COMMAND)
+  unset(INSTALL_COMMAND)
+  unset(BUILD_IN_SOURCE)
+  unset(BUILD_BYPRODUCTS)
+endmacro()
+
+# Macro to create externalproject_add target
+# 
+# Common usage
+#
+# CMAKE_ARGS: cmake(required)
+# PATCH_COMMAND: ALL(optional)
+# CONFIGURE_COMMAND: autoconf(required), meson(required)
+# BUILD_COMMAND: autoconf(required), meson(required), cmake(optional)
+# INSTALL_COMMAND: autoconf(required), meson(required), cmake(optional)
+# BUILD_IN_SOURCE: ALL(optional)
+# BUILD_BYPRODUCTS: ALL(required)
+#
+macro(BUILD_DEP_TARGET)
+  include(ExternalProject)
+
+  if(CMAKE_ARGS)
+    set(CMAKE_ARGS CMAKE_ARGS ${CMAKE_ARGS})
+  endif()
+
+  if(PATCH_COMMAND)
+    set(PATCH_COMMAND PATCH_COMMAND ${PATCH_COMMAND})
+  endif()
+
+  if(CONFIGURE_COMMAND)
+    set(CONFIGURE_COMMAND CONFIGURE_COMMAND ${CONFIGURE_COMMAND})
+  endif()
+
+  if(BUILD_COMMAND)
+    set(BUILD_COMMAND BUILD_COMMAND ${BUILD_COMMAND})
+  endif()
+
+  if(INSTALL_COMMAND)
+    set(INSTALL_COMMAND INSTALL_COMMAND ${INSTALL_COMMAND})
+  endif()
+
+  if(BUILD_IN_SOURCE)
+    set(BUILD_IN_SOURCE BUILD_IN_SOURCE ${BUILD_IN_SOURCE})
+  endif()
+
+  if(NOT BUILD_BYPRODUCTS)
+    set(BUILD_BYPRODUCTS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/${${MODULE}_BYPRODUCT})
+  else()
+    set(BUILD_BYPRODUCTS ${BUILD_BYPRODUCTS})
+  endif()
```

Comment:
you could invert the if for consistency and readablility

Reply:
Will do

Reply:
Have fixed that.
Also changed to BUILD_BYPRODUCTS is an optional. In the case of ffmpeg, we dont set a byproduct (it has more than one essentially), so checks have been put in to handle no byproduct, and to not set the byproduct vars if they arent found in the VERSION file.
This is also the case for MODULE_LIBRARY, if there isnt a byproduct, dont generate that LIBRARY var automatically. This just means the dev has to set the MODULE_LIBRARY var if its required. in the case of ffmpeg, we dont use FFMPEG_LIBRARY, so no need for it to be manually set.